sdk/monitor/Azure.Monitor.Ingestion/api/Azure.Monitor.Ingestion.netstandard2.0.cs

```diff
@@ -19,11 +21,18 @@ public enum ServiceVersion
             V2021_11_01_Preview = 1,
         }
     }
+    public partial class UploadFailedArgs : Azure.SyncAsyncEventArgs
+    {
+        internal UploadFailedArgs() : base (default(bool), default(System.Threading.CancellationToken)) { }
```

Comment:
This should be public and take the properties as parameters.

Reply:
Do we have a `ModelFactory`?  I think it makes more sense to have an entry there rather than make this constructible.

Reply:
FWIW, in Search the event args has a public ctor. Same thing for Service Bus. Model Factory is typically used for output model types.

Reply:
`EventArgs` are effectively output model types.  You never create them as a user.  You just handle an event and they're given to you.

I think it was probably a mistake to leave the .ctors public in Search.  If we already have a `ModelFactory`, I'd consider moving these there.  If we don't, I wouldn't create a `ModelFactory` just for them.